core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -200,6 +202,7 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
     if (zkUtils.pathExists(toResourcePath(resource))) {
       zkUtils.deletePath(toResourcePath(resource))
       updateAclChangedFlag(resource)
+      updateCache(resource, Set())
```

Comment:
Shouldn't we be only updatingCache whenever changes are not made by us. So, in `updateAclChangedFlag` should update last update i (update_id), which can be path returned by `ZkUtils.createSequentialPersistentPath`, Now, `AclChangedNotificationHandler.processNotification` can ignore any updates that were made prior to update_id.